builds/fabric8-che/plugins/keycloak-plugin-server/src/main/java/com/redhat/che/keycloak/server/KeycloakAuthenticationFilter.java

```diff
@@ -42,9 +44,13 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                 || isInternalRequest(authHeader)) {
             LOG.debug("Skipping {}", requestURI);
             chain.doFilter(req, res);
-        } else {
+        } else if (KeycloakUserChecker.matchesUsername(authHeader)) {
             super.doFilter(req, res, chain);
             LOG.debug("{} status : {}", request.getRequestURL(), ((HttpServletResponse) res).getStatus());
+        } else {
```

Comment:
Since it's a question of taste, I'd prefer letting it like that which, IMO, makes clearer what the change it (since it doesn't change the structure of the control flow but just adds a condition).